lib/lib-search/src/main/java/org/sagebionetworks/search/CloudSearchClient.java

```
@@ -83,46 +67,44 @@ public String getDocumentServiceEndpoint() {
 	}
 
 	public void sendDocuments(String documents) throws ClientProtocolException,
-			IOException, HttpClientHelperException {
-		HttpClientHelper.postContent(httpClient, documentServiceEndpoint,
-				documents, SEND_DOCUMENTS_REQUEST_HEADERS);
-	}
-
-	public void sendDocuments(InputStream stream, long length) throws ClientProtocolException, IOException, HttpClientHelperException {
-		HttpClientHelper.postStream(httpClient, documentServiceEndpoint,
-				stream, length, SEND_DOCUMENTS_REQUEST_HEADERS);
+			IOException {
+		SimpleHttpRequest request = new SimpleHttpRequest();
+		request.setHeaders(SEND_DOCUMENTS_REQUEST_HEADERS);
+		request.setUri(documentServiceEndpoint);
+		httpClient.post(request, documents);
 	}
 
-	public String performSearch(String searchQuery) throws ClientProtocolException, IOException, HttpClientHelperException {
-		String url = searchServiceEndpoint + "?" + searchQuery;
-		String s = null;
+	public String performSearch(String searchQuery) throws ClientProtocolException, IOException, CloudSearchClientException {
+		String uri = searchServiceEndpoint + "?" + searchQuery;
+		SimpleHttpRequest request = new SimpleHttpRequest();
+		request.setUri(uri);
+		SimpleHttpResponse response;
 		long backoffMs = 100L;
-		HttpClientHelperException e = null;
 		do {
-			try {
-				s = HttpClientHelper.getContent(httpClient, url);
-			} catch (HttpClientHelperException e1) {
-				e = e1;
-				if (e1.getHttpStatus() == 507) {
-					try {
-						Thread.sleep(backoffMs);
-					} catch (InterruptedException e2) {
-						// Continue
-					}
-				} else {
-					// rethrow
-					logger.error("performSearch(): Exception rethrown (url="+(url==null?"null":url)+")");
-					throw(e1);
+			response = httpClient.get(request);
+			if (response.getStatusCode() == 200) {
+				return response.getContent();
+			}
+			if (response.getStatusCode() == 507) {
+				try {
+					Thread.sleep(backoffMs);
```

do not silently catch an interrupted exception.